File: src/main/java/com/example/bookbox/entity/Book.java
package com.example.bookbox.entity;

import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.Table;

@Entity
@Table(name = "book")
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int book_id;
    
    private long isbn;
    private String title;
	private float price;
    
    @ManyToMany
    @JoinTable(name = "book_author", 
    joinColumns = @JoinColumn(name = "book_id"),
    inverseJoinColumns = @JoinColumn(name = "author_id"))
    private Set<Author> authorDetails;

    public int getBook_id() {
        return book_id;
    }

    public long getIsbn() {
        return isbn;
    }

	public String getTitle() {
        return title;
    }

    public float getPrice() {
        return price;
    }

    public Set<Author> getAuthorDetails() {
        return authorDetails;
    }
    
}

File: Frontend/src/types/Book.ts
import { Author } from "./Author";

export class Book {
    public book_id;
    public isbn;
    public title;
    public price;
    public authorDetails;

    constructor(book_id: number, isbn: number, title: string, price: number, authorDetails: Author[]) {
        this.book_id = book_id;
        this.isbn = isbn;
        this.title = title;
        this.price = price;
        this.authorDetails = authorDetails;
    }
}
